StepDefinitions/LanguageTestsStepDefinitions.cs

```diff
@@ -0,0 +1,71 @@
+using MarsSpecFlow_HannahN.Pages;
+using MarsSpecFlow_HannahN.Utils;
+using OpenQA.Selenium.Chrome;
+using System;
+using TechTalk.SpecFlow;
+using OpenQA.Selenium.Interactions;
+using OpenQA.Selenium.Support.UI;
+
+namespace MarsSpecFlow_HannahN.StepDefinitions
+{
+    [Binding]
+    internal class LanguageTestsStepDefinitions : CommonDriver 
+    {
+        LoginPage loginPageObj = new LoginPage();
+        LanguagePage languagePageObj = new LanguagePage();
+        [Given(@"user login Mars")]
+        public void GivenUserLoginMars()
+        {
+            driver = new ChromeDriver();
+            loginPageObj.loginPage(driver);
+        }
+
+        [When(@"user add a new language")]
+        public void WhenUserAddANewLanguage()
+        {
+            languagePageObj.addLanguage(driver);
+        }
+
+        [Then(@"new language is added successfully")]
+        public void ThenNewLanguageIsAddedSuccessfully()
+        {
+            languagePageObj.verifyAddLanguage(driver);
+        }
+
+        [When(@"user cancel update language")]
+        public void WhenUserCancelUpdateLanguage()
+        {
+            languagePageObj.cancelUpdateLanguage(driver);
```

Comment:
Please do not pass the driver In action methods (methods which in pages doing things like create, delete etc ) instead get the pages to extend to common driver where you will create a common driver class which will initialise the web driver 

public class CommonDriver
    {
        public static IWebDriver driver;

        public void initialise()
        {
            // create driver
            driver = new ChromeDriver();

        }

    }

Reply:
fix this with all action methods